### PR TITLE
Update PHPUnit to 5.7 and 6.0. Update QA tools

### PR DESCRIPTION
**Pull request by demiankatz**
This pull request is an attempt to begin bringing this project in line with the newest QA standards of other ZF components. Work is based in part on https://github.com/zendframework/zend-http/pull/115.

@Xerkus: this is as far as I was able to get today. I think this could be merged as-is, as it is useful progress, or we can leave this PR open and I can create a TODO checklist to get through all of the various steps necessary to bring the project up to date. Please let me know how you would like to proceed -- a series of incremental PR's, or one big one.

### CODE REVIEW

File: tests/Bootstrap.php [hunk 1]
```diff
@@ -13,12 +13,14 @@
  */
 error_reporting(E_ALL | E_STRICT);
 
-$phpUnitVersion = PHPUnit_Runner_Version::id();
-if ('@package_version@' !== $phpUnitVersion && version_compare($phpUnitVersion, '3.5.0', '<')) {
-    echo 'This version of PHPUnit (' . PHPUnit_Runner_Version::id() . ') is not supported in Zend Framework 2.x unit tests.' . PHP_EOL;
-    exit(1);
+if (class_exists('PHPUnit_Runner_Version', true)) {
```
Comment (Xerkus) [inline tests/Bootstrap.php]: This check can safely be dropped

File: tests/ZendService/Amazon/Ec2/AbstractTest.php [hunk 2]
```diff
@@ -32,7 +33,7 @@ public function testSetRegion()
     public function testSetInvalidRegionThrowsException()
     {
         $ec2 = new TestAmazonAbstract('TestAccessKey', 'TestSecretKey');
-        $this->setExpectedException(
+        $this->expectException(
```
Comment (Xerkus) [inline tests/ZendService/Amazon/Ec2/AbstractTest.php]: `setExpectedException` was replaced by `expectException()` + `expectExceptionMessage()` in phpunit

File: tests/ZendService/Amazon/Ec2/AbstractTest.php [hunk 3]
```diff
@@ -32,7 +33,7 @@ public function testSetRegion()
...
             'ZendService\Amazon\Ec2\Exception\InvalidArgumentException',
```
Comment (Xerkus) [inline tests/ZendService/Amazon/Ec2/AbstractTest.php]: Strings with class names should be replaced with ::class and imported with `use` statements across all codebase for the benefit of static analysis tools.
Might be done as separate QA related PR along with coding styles fixes.

**Comment by demiankatz** [comment]
@Xerkus, I believe I have taken care of all of your suggestions. If you would like me to rebase this so that all of the PHPUnit stuff is in a single commit, let me know and I'll be happy to take care of it tomorrow. Heading home for now! Thanks again for your help.

**Comment by Xerkus** [comment]
I pushed few commits adding new travis config, composer.json scripts and bumping zend-http, which is causing problems at its lowest versions  
php-cs-fixer is to be replaced in next PR for `squizlabs/php_codesniffer`, so `cs-check` and `cs-fix` scripts are essentially placeholders.
`license-check` script is a placeholder for `malukenho/docheader` tool to be introduced in following PRs

zend-http bump is causing merge conflicts, please rebase.

**Comment by demiankatz** [comment]
@Xerkus, I have rebased this onto zendframework:develop. Please let me know if you would like me to do any further work on this, or if you prefer to take it from here. If you'd like to split up the remaining work, just let me know what steps I should work on.

**Comment by Xerkus** [comment]
@demiankatz thanks for your contribution.

Another QA changes remaining are getting rid of obsolete php-cs-fixer and introducing docheader. Would you like to do that too?

**Comment by demiankatz** [comment]
@Xerkus,

I started looking at docheader yesterday, but I got a bit confused when I realized that the tool seems to check the headers, but does not offer a way to automatically update them. Is it necessary to manually update all of the headers and only use docheader to remind us that changes are needed, or am I missing some tool to automatically make the comments uniform?

In any case, I'm happy to do the docheader and php-cs-fixer work; I will begin on php-cs-fixer right now. I'll wait to hear from you about docheader before proceeding on that point, since I don't want to waste time on manual effort if I don't have to (though I'm willing to do it if there's no better way).

thanks,
Demian

**Comment by froschdesign** [comment]
@demiankatz 

> …but does not offer a way to automatically update them.

At the moment this is still a [feature request](https://github.com/malukenho/docheader/issues/7).

**Comment by demiankatz** [comment]
@froschdesign, good to know -- thanks for the clarification! That will definitely be a very valuable feature if implemented, but I'll deal with this manually in the meantime.

**Pull request merged**